src/compiler/scala/tools/nsc/symtab/classfile/Pickler.scala

```diff
@@ -54,6 +54,7 @@ abstract class Pickler extends SubComponent {
       syms.foreach { sym =>
         pickle.putDecl(sym)
       }
+      pickle.writeArray()
```

Comment:
ok, I'm probably being thick, but my understanding of this code is that right after the argument lambda returns, we call `pickle.close()`, which immedately calls `writeArray`; what does moving it out here do?

Reply:
It's only different when whe exit with a `FatalError` thrown by `putType` before we get to this line. `close()` is called in a `finally`, and if that calls `writeArray` a cascading `NoSuchElementException` happens, which _isn't_ caught and turned into the `erroneous or inacessible type` error in `PicklerPhase.apply`.

Reply:
Fascinating, I guess. Makes sense.